R/data_module.R

```diff
@@ -0,0 +1,40 @@
+#' Data module
+#'
+#' Data input for `teal::init` in form of a module
+#'
+#' @param ui (`function(id)`)\cr
+#'  `shiny` `ui` module with `id` argument
+#' @param server (`function(id)`)\cr
+#'  `shiny` server function with `id` as argument. Module should return reactive `teal_data`.
+#' @examples
+#' data <- data_module(
+#'   ui = function(id) {
+#'     ns <- NS(id)
+#'     actionButton(ns("submit"), label = "Load data")
+#'   },
+#'   server = function(id) {
+#'     moduleServer(id, function(input, output, session) {
+#'       eventReactive(input$submit, {
+#'         data <- within(
+#'           teal.data::teal_data(),
+#'           {
+#'             dataset1 <- iris
+#'             dataset2 <- mtcars
+#'           }
+#'         )
+#'         teal.data::datanames(data) <- c("iris", "mtcars")
+#'
+#'         data
+#'       })
+#'     })
+#'   }
+#' )
+#' @export
+data_module <- function(ui, server) {
+  checkmate::assert_function(ui, args = "id", nargs = 1)
+  checkmate::assert_function(server, args = "id", nargs = 1)
+  structure(
+    list(ui = ui, server = server),
+    class = "data_module"
```

Comment:
`teal_data_module`? That would fit nicely to the `teal_data` class name.

Reply:
I would opt for `teal_module_data` to be consistent with `teal_module_landing` and `teal_module_previewer`. Unless this is not an extension of `team_module`.

Reply:
> I would opt for `teal_module_data` to be consistent with `teal_module_landing` and `teal_module_previewer`. Unless this is not an extension of `team_module`.

Disagree. This class is different to the one you mentioned. First of all it does not inherit from `teal_module`. Second of all this is not meant to be passed as `module` argument value of `init()` - it is meant to be a `data` argument value instead. IMHO we should keep them separate to avoid confusion.

Reply:
We have some unwritten convention:
- Analytical modules have a prefix `tm_`

And we have additional two special modules which are "inheriting" from `"teal_module"`:
- "teal_module_landing" produced by `landing_popup_module()`
- "teal_module_previewer" produced by `reporter_previewer_module()`

I support `teal_data_module`, do we all agree here?

Reply:
I prefer `teal_data_module` over `teal_module_data`

Reply:
Like I said, unless. I am pretty sure they will be confused anyway. Nomenclature throughout NEST is not entirely consistent and this difference will likely be seen as an inconsistency rather than a sign of a different entity.

I don't mind `teal_data_module` as such.